[[ Tests ]] Add extension testing capabilities

livecodeali opened the pull request:
This patch adds extension testing to our continuous integration, by adding a further check target in the
extensions directory. At some point the tests/Makefile will need tweaking to make sure that only
<extension>/tests/ directories are searched for suitable test handlers.

Also TestLoadExtension has been added to the test library to facilitate loading of extensions in the
TestSetup handler. At some point it might make sense for extension tests to automatically load themselves
but for now it seems fine to have them do it explicitly.

Note this does no dependency analysis, so it is currently not possible to test extensions that rely on modules
other than those built in to the engine.

Commit:
[[ Tests ]] Add extension testing capabilities

This patch adds extension testing to our continuous integration, by adding a further check target in the
extensions directory. At some point the tests/Makefile will need tweaking to make sure that only
<extension>/tests/ directories are searched for suitable test handlers.

Also TestLoadExtension has been added to the test library to facilitate loading of extensions in the
TestSetup handler. At some point it might make sense for extension tests to automatically load themselves
but for now it seems fine to have them do it explicitly.

Note this does no dependency analysis, so it is currently not possible to test extensions that rely on modules
other than those built in to the engine.

Commit:
[[ Tests ]] Make sure initial engine path is searched for extensions folder

peter-b commented:
@livecodeali Why doesn't `TestLoadExtension` take the "full" name of the extension as its argument?

livecodeali commented:
@peter-b because we unzip them to a bespoke place. I guess it might be better to pass in the full name and use the last item to find them in the bespoke place.

Commit:
[[ Tests ]] Pass in full name of extension to test lib

Even though only the last item is used. If we ever change the way the
extensions are unzipped to do the proper thing, at least only the test
lib has to be changed.

peter-b commented:
@livecodeali I sort of feel like it might be better to pass in the full name and use the last item, just so it's clearer what the intention is (you can easily throw an error if a test tries to load an extension that the test library doesn't know how to load).

peter-b commented:
@livecode-vulcan review ok 18fddec

livecode-vulcan commented:
💙 review by @peter-b ok 18fddecaa211a08819efb451112203fbaf2646bd

livecode-vulcan commented:
😎 test success 18fddecaa211a08819efb451112203fbaf2646bd
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1074)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1083)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1071)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1076)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1087)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1092)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1092)

mwieder commented:
Note that the copyright notice still says "Copyright (C) 2015 Runtime Revolution Ltd."

peter-b commented:
Hah, good spot

mwieder commented:
I see there's a "packaged_extensions" folder referenced. Is that where the built-in extensions are going to end up?

livecodeali commented:
That's where they are unzipped to in the course of building everything from source, yes.

mwieder commented:
OK - so this isn't a change in where the installer will place extensions on the user's system then?

peter-b commented:
@mwieder No change (well, not in this pull request at least)

livecodeali commented:
@mwieder Plus, when we fix your bug about revEnvironmentExtensionsPath and UserExtensionsPath, you won't need to worry about it ;-)

mwieder commented:
Hooray! on all counts.
